tests/unit_tests/test_lib/test_hashutil.py

```diff
@@ -87,3 +90,39 @@ def test_md5_file_hex_three_files(data1, text, data2):
     # Intentionally provide the paths out of order (check sorting).
     path_hash = hashutil.md5_file_hex("c.bin", "a.bin", "b.txt")
     assert hashlib.md5(data).hexdigest() == path_hash
+
+
+@pytest.mark.parametrize(
+    "filesize",
+    [
+        1024,
+        8192,
+        2 * 1024 * 1024,
+    ],
+)
+def test_md5_file_hashes_on_mounted_filesystem(filesize, tmp_path, fs: FakeFilesystem):
```

Comment:
Interestion, `pyfakefs` manages to reproduce this error (i.e. this fails without the fix)? I'm not shocked but a little surprised.

Reply:
I'm surprised too, as `pyfakefs` is pretty finicky.  It does reproduces the OSError but for some reason gets the `errno` wrong.
- `pyfakefs` -> raises OSError with "Invalid argument" i.e. 22 (errno.EINVAL)
- original error -> raises OSError with "No such device" i.e. 19 (errno.ENODEV)

Haven't figured out if there's extra configuration I need to fully reproduce the same errno here, but it's the closest I seem to get in the test environment right now.  At any rate -- once I add in the try/except block from [your suggestion](https://github.com/wandb/wandb/pull/8473#pullrequestreview-2325796597) -- it'll broadly catch any OSError, so this seems acceptable